pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -360,3 +332,15 @@ func (m *Matrix) validateParameterInOneOfMatrixOrParams(params []Param) (errs *a
 	}
 	return errs
 }
+
+func (m *Matrix) validateNoWholeArrayResults() (errs *apis.FieldError) {
+	if m.HasParams() {
+		for i, param := range m.Params {
+			val := param.Value.StringVal
+			if strings.Contains(val, "results") && strings.Contains(val, "[*]") {
```

Comment:
this will probably match strings we don't necessarily want to, e.g. imagine a param named "results1234"; this will match `$(params.results1234[*])`. Can you see whether we have existing parsing functions in the codebase to determine if a string is a reference to an array result?

Reply:
Updated to leverage existing `LooksLikeContainsResultRefs()` function